draft-ietf-anima-brski-prm.md

```diff
@@ -1238,48 +1268,60 @@ In {{!RFC8995}}, the pledge generates PVR as CMS-signed JSON and PER as PKCS#10
+In addition, the registrar MUST verify that
 
-The HTTP response Content-Type header field is set to `application/voucher-jws+json` as defined in {{I-D.ietf-anima-jws-voucher}} if no content negotiation is used.
+* the `agent-provided-proximity-registrar-cert` field of the PVR contains the registrar-own EE certificate to ensure the registrar in proximity of the Registrar-Agent is the desired registrar for this PVR.
+* the `agent-signed-data` field of the PVR is signed with the private key corresponding to the Registrar-Agent EE certificate;
+  this is done via the SubjectKeyIdentifier in the `kid` Header Parameter of the JWS Protected Header of the `agent-signed-data` field;
+  the registrar MAY use the Registrar-Agent EE certificate verified during TLS client authentication;
+  otherwise the Registrar-Agent EE certificate(s) need to be provided via configuration or a repository.
+* the product-serial-number inside the `agent-signed-data` matches the `serial-number` field of the PVR as well as the X520SerialNumber field of the pledge IDevID certificate in the JWS Protected Header of the PVR.
+* the Registrar-Agent EE certificate is still valid;
+  this is necessary to avoid that a rogue Registrar-Agent generates `agent-signed-data` objects to onboard arbitrary pledges at a later point in time, see also {{sec_cons_reg-agt}}.
 
+If the registrar is unable to process the request or validate the PVR, it SHOULD respond with an HTTP error code.
+The following client error codes SHOULD be used:
 
-### Request Artifact: Pledge Voucher-Request (PVR)
+* 400 Bad Request: if the pledge detects an error in the format of the request
+* 403 Forbidden: if the registrar detected that one or more security related fields are not valid or if the pledge-provided information could not be used with automated allowance
+* 406 Not Acceptable: if the Accept request header field indicates a type that is unknown or unsupported
+* 415 Unsupported Media Type: if the Content-Type request header field indicates a type that is unknown or unsupported
 
-For BRSKI-PRM, the Registrar-Agent sends the PVR by HTTP POST to the same registrar endpoint as introduced by BRSKI: "/.well-
-known/brski/requestvoucher", but with a Content-Type header field for JSON-in-JWS".
+Otherwise, the registrar converts the PVR artifact to an RVR artifact as defined in {{rvr-artifact}}.
+It then establishes a TLS session with mutual authentication to the MASA of the pledge according to {{Section 5.4 of !RFC8995}} and requests a voucher from the MASA according to {{Section 5.5 of !RFC8995}}.
 
-After receiving the PVR from Registrar-Agent, the registrar SHALL perform the verification as defined in {{Section 5.3 of RFC8995}}.
-In addition, the registrar SHALL verify the following parameters from the PVR:
+TODO: Needs MASA modifications for JWS Voucher. It could work with existing MASAs if RVR were CMS.
 
```

Comment:
Yes, MASA needs modification to use JWS instead of CMS. But this is only related to the form factor, not the general approach.

Reply:
My thought was that modifications to a deployed BRSKI system could be minimized by keeping the messages to the MASA exactly the same.

Yet we probably cannot assume that the registrar always supports both formats and could convert for the MASA, right? (only a combined BRSKI + BRSKI-PRM registrar could do so).
However, if a manufacturer produced a device running BRSKI-PRM, it should be expected that they also have a MASA providing BRSKI-PRM features... Hence, a registrar facilitating BRSKI-PRM can rely on the corresponding MASA to understand the new format. Yes?

Reply:
Yes, the manufacturer build the device and knows the method implemented in the pledges for onboarding.